### ISSUE TITLE
Clicking on the worker/fighter button should be permitted more than once per turn.

### ISSUE THREAD

**Issue by Bertram25** [OpenDungeons/OpenDungeons#311]
If possible in lan, but overall, would be good in single-player.

**Comment by akien-mga** [comment]
I've just noticed that a part of this issue is not fixed yet: you can now click several times per turn on a worker/fighter button, and it grabs as many creatures as you clicked, but the visual feedback of the button is wrong: it gets pressed only once per turn visually.

**Comment by hwoarangmy** [comment]
It is probably a cegui issue. I will let @Bertram25 have a look as he is more used to that than me

**Comment by Bertram25** [comment]
Ok, I'll look at that.

**Comment by Bertram25** [comment]
First things first, I've just discovered here:
http://www.ogre3d.org/tikiwiki/Basic+Tutorial+4

That if we are doing one logic update per frame update, we should do it in FrameListener::frameRenderingQueued() instead of FrameListener::frameStarted() for performance reason. Quoting:

> That of course depends on what you need to do, but if you only want to update your stuff once per frame, put it in the frameRenderingQueued event, because that one is called just before the GPU is made busy by flipping the render buffer.
> So you want to keep your CPU busy while the GPU works. 
> 
> Of course because the frame's rendering
> commands have already been issued, any changes you make will only
> take effect from the next frame, but in most cases that's not noticeable.

This might be worth trying IMHO.

**Comment by Bertram25** [comment]
I've looked at this carefully. I do think our logic is sane and that the gui update might lag a bit more in game because of lower FPS there and maybe because of the gui responding to double-click event calls. I'll try the solution given above to see whether we earn a bit of FPS, but the issue should be closed as wont fix for the rest.
It is at least my opinion. Feel free to prove me wrong (I would be glad to learn cegui performance fixes here) but this is what I'll do before closing.

**Comment by Bertram25** [comment]
I'll request to close this one as wontfix or later. We need more FPS before dealing with GUI performance.

**Comment by akien-mga** [comment]
I'm not sure I really understand why it would be linked to our FPS performance? I currently run OD with around 35 FPS, but the issue I mentioned seems to be that the "clicked" state of the grab button is only shown once per turn (at least it looks like it), even if it is clicked more than once per turn.

But I'm fine with closing this one for now (the issue title refers to something that has indeed been fixed), we can always open a new issue for the Later milestone if need be.

**Comment by Bertram25** [comment]
My point is more about seeing how the OD gui behave at 60 FPS. But you might be right, we might have to disable double-click event GUI update in order to get the desired effect. I sincerely don't know how yet, though :)

**Comment by Bertram25** [comment]
I'll close this one. #343 is for 0.5.0 :)
